content/en/docs/armory-agent/agent-mtls.md

```diff
@@ -0,0 +1,267 @@
+
+```bash
+kubectl create secret tls <your-secret-name> \
+   --cert=clouddriver.crt --key=clouddriver.key  
+```
+
+Or:
+
+```bash
+kubectl create secret generic <your-secret-name> \
+   --from-file=clouddriver.crt --from-file=clouddriver.key
+```
+
+### Mount the secret
+
+Mount the secret in the plugin.
+
+In your `agent-plugin/clouddriver-plugin.yaml` file,
+`spec.kustomize.clouddriver.deployment.patchesStrategicMerge` section,
+add the following lines to mount the Clouddriver cert from your secret:
+
+{{< prism lang="yaml" line="20-21, 26-28" >}}
+spec:
+  kustomize:
+    clouddriver:
+      deployment:
+        patchesStrategicMerge:
+          - |
+            spec:
+              template:
+                spec:
+                  initContainers:
+                  - name: kubesvc-plugin
+                    image: docker.io/armory/kubesvc-plugin:<release>
+                    volumeMounts:
+                      - mountPath: /opt/plugin/target
+                        name: kubesvc-plugin-vol
+                  containers:
+                  - name: clouddriver
+                    volumeMounts:
+                      - mountPath: /opt/clouddriver/lib/plugins
+                        name: kubesvc-plugin-vol
+                      - mountPath: /opt/clouddriver/cert  
+                        name: cert
+                  volumes:
+                  - name: kubesvc-plugin-vol
+                    emptyDir: {}
+                  - name: cert
+                    secret:
+                      secretName: <your-secret-name>
+{{< /prism >}}
+
+
+### Configure the plugin
+
+Configure the plugin to use the mounted certs in the `agent-plugin/config.yaml` file. Note that `certificateChain`, `trustCertCollection`, and `privateKey` values must in `file:///filepath/filename` format.
+
+{{< prism lang="yaml" line="12-20" >}}
+apiVersion: spinnaker.armory.io/{{< param "operator-extended-crd-version" >}}
+kind: SpinnakerService
+metadata:
+  name: spinnaker
+spec:
+  spinnakerConfig:
+    profiles:
+      clouddriver:
+        # See https://docs.armory.io/docs/installation/armory-agent/agent-options/
+        kubesvc:
+          cluster: redis
+          grpc:
+            server:
+              security:
+                enabled: true
+                certificateChain: file:<path-to-agent.crt>
+                trustCertCollection: file:<path-to-agent.crt>
+                privateKey: file:<path-to-agent.key>
+                clientAuth: REQUIRE
+{{< /prism >}}
+
+See the {{< linkWithTitle "agent-plugin-options.md" >}} page for additional options.
+
+## Agent service configuration
+
+### Create a secret
+
+Create a secret in the target cluster namespace where the Agent resides.
+
+```bash
+kubectl create secret generic <agent-secret-name> \
+--from-file=agent.crt --from-file=agent.key
+```
+
+### Modify deployment configuration
+
+Modify the Agent's deployment configuration to mount the certs. @TODO what file is this in?
```

Comment:
what file is this in?

Reply:
According to the existing installation doc: https://docs.armory.io/docs/armory-agent/armory-agent-quick/#kustomize that file is inside the referenced tarball (https://armory.jfrog.io/artifactory/manifests/kubesvc/armory-agent-0.5.11-kustomize.tar.gz)